src/Sylius/Bundle/ApiBundle/Serializer/CommandFieldItemIriToIdentifierDenormalizer.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Bundle\ApiBundle\Converter\CommandFieldItemIriToIdentifierStorageInterface;
+use Sylius\Bundle\ApiBundle\Converter\ItemIriToIdentifierConverter;
+use Sylius\Bundle\ApiBundle\DataTransformer\CommandAwareInputDataTransformer;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class CommandFieldItemIriToIdentifierDenormalizer implements ContextAwareDenormalizerInterface
+{
+    /** @var DenormalizerInterface */
+    private $objectNormalizer;
+
+    /** @var ItemIriToIdentifierConverter */
+    private $itemIriToIdentifierConverter;
+
+    /** @var CommandAwareInputDataTransformer */
+    private $commandAwareInputDataTransformer;
+
+    /** @var ServiceLocator */
+    private $commandFieldIriToIdentifierStorages;
+
+    public function __construct(
+        DenormalizerInterface $objectNormalizer,
+        ItemIriToIdentifierConverter $itemIriToIdentifierConverter,
+        CommandAwareInputDataTransformer $commandAwareInputDataTransformer,
+        ServiceLocator $commandFieldIriToIdentifierStorages
+    ) {
+        $this->objectNormalizer = $objectNormalizer;
+        $this->itemIriToIdentifierConverter = $itemIriToIdentifierConverter;
+        $this->commandAwareInputDataTransformer = $commandAwareInputDataTransformer;
+        $this->commandFieldIriToIdentifierStorages = $commandFieldIriToIdentifierStorages;
+    }
+
+    public function supportsDenormalization($data, $type, $format = null, array $context = [])
+    {
+        /** @var string $commandFieldItemIriToIdentifierStorageName */
+        foreach ($this->getCommandFieldIriToIdStoragesNames() as $commandFieldItemIriToIdentifierStorageName) {
+            /** @var CommandFieldItemIriToIdentifierStorageInterface $commandFieldItemIriToIdentifierConverter */
+            $commandFieldItemIriToIdentifierConverter = $this
+                ->commandFieldIriToIdentifierStorages
+                ->get($commandFieldItemIriToIdentifierStorageName)
+            ;
+
+            if ($this->getInputClass($context) === $commandFieldItemIriToIdentifierConverter->getCommandType()) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
This basically looks like:

```suggestion
        foreach ($this-commandFieldIriToIdentifierStorages as $name => $storage) {
            if ($this->getInputClass($context) === $storage->getCommandType()) {
                return true;
            }
        }

        return false;
```